stable/cosi-bucket-kit/templates/job-readiness.yaml

```
@@ -0,0 +1,188 @@
+                    "insecure_skip_verify": false
+                  },
+                  "trace": {
+                    "enable": false # Enable to debug errors (if any)
+                  },
+                  "part_size": 134217728
+                }
+              }' <<< "$bucketInfo" > "$tmpfile"
+              echo "Transformed bucketInfo to federated-store.yaml. Updating federated-store secret..."
+
+              kubectl create secret generic federated-store -n {{ .Release.Namespace }} --from-file=federated-store.yaml="$tmpfile" --dry-run=client -o yaml | kubectl apply -f -
+              kubectl label secret federated-store -n {{ .Release.Namespace }} app.kubernetes.io/kommander-kubecost-federated-store=true --overwrite
+              kubectl annotate secret federated-store -n {{ .Release.Namespace }} app.kubernetes.io/kommander-kubecost-federated-store-processed=true --overwrite
+
+              kubectl create configmap kubecost-object-store-config -n {{ .Release.Namespace }} --save-config --from-literal=objectStoreStatus=ready --dry-run=client -o yaml | kubectl apply -f -
+              rm "$tmpfile"
+        {{- end }}
+        {{- if .Values.cosiBucketKit.transformations.harbor.enabled }}
+        - name: transform-harbor-cosi-secret
+          image: {{ .Values.cosiBucketKit.transformations.kubectlImage }}
+          command:
+            - bash
+            - -c
+            - |
+              set -o nounset
+              set -o pipefail
+
+              echo() {
+                  command echo $(date) "$@"
+              }
+              {{- if gt (len .Values.cosiBucketKit.bucketAccesses) 1 }}
+              {{- fail "Error: .Values.cosiBucketKit.bucketAccesses array size must not exceed 1 if harbor transformation is enabled." }}
+              {{- end }}
+
+              {{- $cmName := .Values.cosiBucketKit.transformations.harbor.cmName }}
+              {{- $ns := .Release.Namespace -}}
+              {{- range .Values.cosiBucketKit.bucketAccesses }}
+              # Update the cosi secret with harbor specific keys.
+              kubectl create secret generic {{ .credentialsSecretName }} -n {{ $ns }} \
+                --from-literal=REGISTRY_STORAGE_S3_ACCESSKEY=$(kubectl get secret {{ .credentialsSecretName }} -n {{ $ns }} -o jsonpath="{.data.BucketInfo}" | base64 --decode | jq -r '.spec.secretS3.accessKeyID') \
+                --from-literal=REGISTRY_STORAGE_S3_SECRETKEY=$(kubectl get secret {{ .credentialsSecretName }} -n {{ $ns }} -o jsonpath="{.data.BucketInfo}" | base64 --decode | jq -r '.spec.secretS3.accessSecretKey') \
+                --dry-run=client -o yaml | kubectl apply -f -
+              # Create a configmap with the name of the secret from above.
+              kubectl create configmap {{ $cmName }} -n {{ $ns }} \
+                --from-literal="values.yaml=$(printf -- '---\npersistence:\n  imageChartStorage:\n    s3:\n      existingSecret: {{ .credentialsSecretName }}')" \
```

```suggestion
                --from-file=values.yaml=<(cat <<EOF
                persistence:
                  imageChartStorage:
                    type: s3
                    s3:
                      existingSecret: cosi-harbor-bucket-credentials
                EOF
                ) \
```

We need to override the default `filesystem` storage type.